File: csharp/lib/Property/SlamDataSupportedProperty.cs
/*
 * SPDX-FileCopyrightText: 2017-2023 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

using MessagePack;

namespace SensCord
{
    /// <summary>
    /// Data format supported by SLAM stream.
    /// </summary>
    [MessagePackObject]
    public class SlamDataSupportedProperty : IBaseProperty
    {
        /// <summary>
        /// Property key.
        /// </summary>
        public static readonly string ConstKey = "slam_data_supported_property";

        /// <summary>
        /// Property key.
        /// </summary>
        [IgnoreMember]
        public string Key { get; } = ConstKey;

        /// <summary>
        /// Odometry supported.
        /// </summary>
        [Key("odometry_supported")]
        public bool OdometrySupported { get; set; }

        /// <summary>
        /// GridMap supported.
        /// </summary>
        [Key("gridmap_supported")]
        public bool GridMapSupported { get; set; }

        /// <summary>
        /// PointCloud supported.
        /// </summary>
        [Key("pointcloud_supported")]
        public bool PointCloudSupported { get; set; }

        /// <summary>
        /// To string.
        /// </summary>
        /// <returns>A string that represents the current object.</returns>
        public override string ToString()
        {
            return string.Format(
                "{0}(odometry={1}, gridmap={2}, pointcloud={3})",
                this.Key, this.OdometrySupported, this.GridMapSupported, this.PointCloudSupported);
        }
    }
}

File: java/senscord/src/senscord_properties/SlamDataSupportedProperty.java
/*
 * SPDX-FileCopyrightText: 2019-2023 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package senscord_properties;

import com.sun.jna.Structure;
import java.util.Arrays;
import java.util.List;

/**
 * Data format supported by SLAM stream.
 * 
 * @see "senscord::SlamDataSupportedProperty"
 */
public class SlamDataSupportedProperty extends Structure {
    public byte odometry_supported;
    public byte gridmap_supported;
    public byte pointcloud_supported;

    public static class ByReference extends SlamDataSupportedProperty
            implements Structure.ByReference {
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("odometry_supported", "gridmap_supported", "pointcloud_supported");
    }


    /**
     * Get position and attitude data supported or not.
     * 
     * @return position and attitude supported or not
     */
    public byte GetOdometrySupported() {
        return odometry_supported;
    }

    /**
     * Get GridMap supported or not.
     * 
     * @return GridMap supported or not
     */
    public byte GetGridMapSupported() {
        return gridmap_supported;
    }

    /**
     * Get PointCloud supported or not.
     * 
     * @return PointCloud supported or not
     */
    public byte GetPointCloudSupported() {
        return pointcloud_supported;
    }

    /**
     * Set position and attitude data supported or not.
     * 
     * @param odometry_supported position and attitude data supported or not
     */
    public void SetOdometrySupported(byte odometry_supported) {
        this.odometry_supported = odometry_supported;
    }

    /**
     * Set GridMap supported or not.
     * 
     * @param gridmap_supported GridMap supported or not
     */
    public void SetGridMapSupported(byte gridmap_supported) {
        this.gridmap_supported = gridmap_supported;
    }

    /**
     * Set PointCloud supported or not.
     * 
     * @param pointcloud_supported PointCloud supported or not
     */
    public void SetPointCloudSupported(byte pointcloud_supported) {
        this.pointcloud_supported = pointcloud_supported;
    }
}
